File: source/frontend/src/lib/dto/response/entity/InfluenceResponseDto.ts
import {
    ArchitectureRequirementResponseDto,
    fromArchitectureRequirementResponseDto
} from "$lib/dto/response/entity/ArchitectureRequirementResponseDto";
import {Influence} from "$lib/domain/entity/ad/Influence.svelte";

export class InfluenceResponseDto {
    id : string | undefined;
    value : number | undefined;
    architectureRequirement : ArchitectureRequirementResponseDto | undefined;

    static create() {
        return new InfluenceResponseDto();
    }
}

export function fromInfluenceListResponseDto(influenceListResponseDto : InfluenceResponseDto[]) : Influence[] {
    return influenceListResponseDto.map(fromInfluenceResponseDto);
}

export function fromInfluenceResponseDto(influenceResponseDto : InfluenceResponseDto) : Influence {
    const influence : Influence = Influence.create();

    influence.id = influenceResponseDto.id ?? "";
    influence.value = influenceResponseDto.value ?? 0.0;

    if (influenceResponseDto.architectureRequirement) {
        influence.architectureRequirement = fromArchitectureRequirementResponseDto(influenceResponseDto.architectureRequirement);
    }

    return influence;
}

File: source/backend/src/main/java/ch/unisg/backend/controller/http/dto/response/ad/InfluenceResponseDto.java
package ch.unisg.backend.controller.http.dto.response.ad;

import ch.unisg.backend.controller.http.dto.response.ResponseDto;
import ch.unisg.backend.controller.http.dto.response.ar.ArchitectureRequirementResponseDto;
import ch.unisg.backend.core.domain.entities.classes.ad.Influence;

import java.util.HashMap;
import java.util.List;

public class InfluenceResponseDto extends ResponseDto {
    public static List<HashMap<String, Object>> toJson(List<Influence> influenceList) {
        return influenceList.stream().map(InfluenceResponseDto::toJson).toList();
    }

    public static HashMap<String, Object> toJson(Influence influence) {
        HashMap<String, Object> result = new HashMap<>();

        result.put("id", influence.getId());
        result.put("value", influence.getValue());
        result.put("architectureRequirement", ArchitectureRequirementResponseDto.toJson(influence.getArchitectureRequirement()));

        return result;
    }

}
